zerver/views/streams.py

```diff
@@ -324,6 +324,10 @@ def update_stream_backend(
     if proposed_is_private and proposed_is_default_stream:
         raise JsonableError(_("A default channel cannot be private."))
 
+    # Ensure that a moderation request channel isn't set to public.
+    if not proposed_is_private and user_profile.realm.get_moderation_request_channel() == stream:
+        raise JsonableError(_("A moderation request channel cannot be public."))
```

Comment:
Changed this to use the same string that you had elsewhere. Every string is extra work for translators, so only add new ones if they serve a useful purpose to users.